Search results component

yordanalexandrov opened the pull request:
Only for review, will not be merged in this state.

svilenvelikov commented on projects/onto-search/src/lib/onto-search-results/onto-search-results.component.scss:
```diff
@@ -0,0 +1,82 @@
+//.onto-table thead, .onto-table tbody, .onto-table tfoot,
+//[onto-table-header-row], [onto-table-row], [onto-table-footer-row] {
+//  background: inherit;
+//}
```
What is this sorcery?

yordanalexandrov replied:
¯\\_(ツ)_/¯

svilenvelikov commented on projects/onto-search/tslint.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../../tslint.json",
+//  "extends": "../../tslint.json",
```
Why this is commented out?

yordanalexandrov committed:
+ Added option to pass TemplateRef for cell

svilenvelikov commented on projects/onto-search/src/lib/onto-search-results/onto-search-results.component.html:
```diff
@@ -8,7 +8,13 @@
+          <ng-container *ngIf="columnConfig.dataFunction; else cellTemplate">
+            {{columnConfig.dataFunction(row)}}
+          </ng-container>
+          <ng-template #cellTemplate [ngTemplateOutlet]="columnConfig.dataTemplate"
```
good :)

svilenvelikov commented on projects/onto-search/src/lib/onto-search-results/onto-search-results.component.ts:
```diff
@@ -0,0 +1,30 @@
+  displayColumns: string[];
+
+
+  constructor() {
```
removethis empty constructor

yordanalexandrov committed:
+ Added component for search result tests
+ Added route to new component
+ Added cypress tests
+ Added cypress coverage package

yordanalexandrov committed:
Merge branch 'master' into wip-search-results

yordanalexandrov committed:
+ Added component for search result tests
+ Added route to new component
+ Added cypress tests
+ Added cypress coverage package

yordanalexandrov committed:
+ Added component for search result tests
+ Added route to new component
+ Added cypress tests
+ Added cypress coverage package

svilenvelikov commented on cypress/steps/search-results-steps.ts:
```diff
@@ -0,0 +1,37 @@
+export default class SearchResultsSteps {
+  static visit(waitLoader: boolean = true) {
```
remove this `waitLoader: boolean = true`

svilenvelikov replied:
I know there is similar shit in the UI app too, but I'll remove it. :)

svilenvelikov commented on cypress/test/search-results.spec.ts:
```diff
@@ -0,0 +1,34 @@
+describe('Onto Search Results Component', () => {
+  it('Should find the component', () => {
+    SearchResultsSteps.visit();
+    SearchResultsSteps.getOntoSearchResultsComponent();
```
I know that cypress will blow if it's not there, but better add `.should('be.visible')` at least

svilenvelikov commented on cypress/test/search-results.spec.ts:
```diff
@@ -0,0 +1,34 @@
+  it('Should have data loaded', () => {
+    SearchResultsSteps.visit();
+
+    SearchResultsSteps.getDataRows();
```
Maybe verify how many rows there are in the table?

yordanalexandrov committed:
+ Added component for search result tests
+ Added route to new component
+ Added cypress tests
+ Added cypress coverage package

teodossidossev reviewed (approved):
lgtm
